adcc/LazyMp.py

```diff
@@ -122,9 +202,27 @@ def dipole_moment(self, level=2):
             raise NotImplementedError("Only dipole moments for level 1 and 2"
                                       " are implemented.")
 
+    @cached_member_function
     def energy_correction(self, level=2):
         """Obtain the MP energy correction at a particular level"""
-        return self.__cpp.energy_correction(level)
+        if level > 3:
+            raise NotImplementedError(f"MP({level}) energy correction "
+                                      "not implemented.")
+        if level < 2:
+            return 0.0
+        hf = self.reference_state
+        if level == 2 and not self.has_core_occupied_space:
+            terms = [(1.0, hf.oovv, self.t2oo)]
+        elif level == 2 and self.has_core_occupied_space:
+            terms = [(1.0, hf.oovv, self.t2oo),
+                     (2.0, hf.ocvv, self.t2oc),
+                     (1.0, hf.ccvv, self.t2cc)]
+        elif level == 3:
+            terms = [(1.0, hf.oovv, self.td2(b.oovv))]
```

Comment:
So MP(3) on CVS now works just like this? I think you might be missing a case here.

Reply:
MP3 on CVS is wrong anyways... I think we should throw an exception right here instead of returning a wrong result.

Reply:
Definitely!